### ISSUE TITLE
IllegalAccessException while running stage mode on JDK9

### ISSUE THREAD

**Issue by eximius313** [playframework/playframework#8566]
### Play Version (2.5.x / etc)
2.6.15

### API (Scala / Java / Neither / Both)
Java

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
Window 7

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
JDK 9.0.4

### Expected Behavior
No exception is thrown 

### Actual Behavior
```
01:07:13.238 [main      ] [debug] play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: available, true
01:07:13.240 [main      ] [debug] play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable
java.lang.IllegalAccessException: class play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6 cannot access class jdk.internal.misc.Unsafe (in module java.base) because module java.base does not export jdk.internal.misc to unnamed module @35eb4a3b
        at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
        at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:589)
        at java.base/java.lang.reflect.Method.invoke(Method.java:556)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6.run(PlatformDependent0.java:295)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:286)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:197)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:74)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:155)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:200)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetectorFactory$DefaultResourceLeakDetectorFactory.newResourceLeakDetector(ResourceLeakDetectorFactory.java:201)
        at play.shaded.ahc.io.netty.util.HashedWheelTimer.<clinit>(HashedWheelTimer.java:88)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.newNettyTimer(DefaultAsyncHttpClient.java:91)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.<init>(DefaultAsyncHttpClient.java:83)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get$lzycompute(AhcWSModule.scala:47)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:44)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:38)
        at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:85)
        at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:77)
        at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:59)
        at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:61)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:148)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:211)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:182)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
        at com.google.inject.Guice.createInjector(Guice.java:87)
        at com.google.inject.Guice.createInjector(Guice.java:78)
        at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:185)
        at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:137)
        at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:51)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:25)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
01:07:13.240 [main      ] [debug] play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - java.nio.DirectByteBuffer.<init>(long, int): available

```
### Reproducible Test Case
Run stage app on JDK9

**Comment by marcospereira** [comment]
Hey @eximius313, this looks like a Netty problem that was fixed in newer versions.

Could you can try with Play 2.7.0-M4 to check that this is not happening this version? Also, I think we can just update async-http-client version in play-ws, do a new release and it will fix the problem.

**Comment by dwijnand** [comment]
Fixed in https://github.com/playframework/play-ws/pull/290.

**Comment by eximius313** [comment]
Is it fixed only in `2.7.x` or you backborted it also to `2.6.x`?

**Comment by dwijnand** [comment]
That PR is for a fresh release of play-ws that we'll pull into Play 2.6.

**Comment by dadepo** [comment]
I am running into this issue with Play Framework version 2.7.2 running with Java-11. Anything I am doing wrong, or this is a sign of a regression?

**Comment by mkurz** [comment]
@dadepo You can upgrade to 2.7.3 already. However, which Scala version do you use?

**Comment by dadepo** [comment]
@mkurz sure I can...
but in the meantime, I am using Scala `2.11.12`...thoughts?

**Comment by mkurz** [comment]
@dadepo Can you post the exception? Is it exactly the same like in the initial post of this issue?

**Comment by mkurz** [comment]
BTW: I recommend to upgrade to Scala 2.12.8 or even 2.13.0 (which is supported as of Play 2.7.3). Newer Scala versions come with much better support for Java 11.

**Comment by dadepo** [comment]
The exception looks exactly like the initial post. But I have copied part of it below:

```
11:55:47.678 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: available, true
11:55:47.688 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable
java.lang.IllegalAccessException: class play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6 cannot access class jdk.internal.misc.Unsafe (in module java.base) because module java.base does not export jdk.internal.misc to unnamed module @267cde2e
        at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
        at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
        at java.base/java.lang.reflect.Method.invoke(Method.java:558)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6.run(PlatformDependent0.java:334)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:325)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:212)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:162)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:207)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetectorFactory$DefaultResourceLeakDetectorFactory.newResourceLeakDetector(ResourceLeakDetectorFactory.java:201)
        at play.shaded.ahc.io.netty.util.HashedWheelTimer.<clinit>(HashedWheelTimer.java:88)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.newNettyTimer(DefaultAsyncHttpClient.java:97)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.<init>(DefaultAsyncHttpClient.java:87)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get$lzycompute(AhcWSModule.scala:60)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:57)
        at play.api.libs.ws.ahc.AhcWSComponents$class.wsClient(AhcWSComponents.scala:32)
        at ...
...
....
(ProductionApplicationLoader.scala:22)
        at boot.ProductionApplicationLoader.load(ProductionApplicationLoader.scala:16)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:59)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:31)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
```

In the meantime, i will also upgrade to 2.12.8 as you suggested

**Comment by mkurz** [comment]
@dadepo Try to upgrade to 2.13 if possible (again, you need 2.7.3 for it to work). Not sure if all your dependencies are published for 2.13 already, but you will see.

**Comment by dadepo** [comment]
I just upgraded to 2.12.8, still got the error. Here is the fuller stacktrace:

```
12:04:06.911 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - -Dio.netty.noUnsafe: false
12:04:06.911 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - Java version: 11
12:04:06.913 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
12:04:06.915 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
12:04:06.916 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - java.nio.Buffer.address: available
12:04:06.921 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - direct buffer constructor: unavailable
java.lang.UnsupportedOperationException: Reflective setAccessible(true) disabled
        at play.shaded.ahc.io.netty.util.internal.ReflectionUtil.trySetAccessible(ReflectionUtil.java:31)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0$4.run(PlatformDependent0.java:224)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:218)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:212)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:162)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:207)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetectorFactory$DefaultResourceLeakDetectorFactory.newResourceLeakDetector(ResourceLeakDetectorFactory.java:201)
        at play.shaded.ahc.io.netty.util.HashedWheelTimer.<clinit>(HashedWheelTimer.java:88)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.newNettyTimer(DefaultAsyncHttpClient.java:97)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.<init>(DefaultAsyncHttpClient.java:87)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get$lzycompute(AhcWSModule.scala:60)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:57)
        at play.api.libs.ws.ahc.AhcWSComponents.wsClient(AhcWSComponents.scala:32)
        at play.api.libs.ws.ahc.AhcWSComponents.wsClient$(AhcWSComponents.scala:29)
        at boot.AppPortalBuiltInComponentsFromContext.wsClient$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.wsClient(ProductionApplicationLoader.scala:21)
        at boot.modules.ExternalClientsComponents.authorisationServiceClient(ExternalClientsComponents.scala:49)
        at boot.modules.ExternalClientsComponents.authorisationServiceClient$(ExternalClientsComponents.scala:46)
        at boot.AppPortalBuiltInComponentsFromContext.authorisationServiceClient$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.authorisationServiceClient(ProductionApplicationLoader.scala:21)
        at boot.modules.ServicesComponents.userService(ServicesComponents.scala:60)
        at boot.modules.ServicesComponents.userService$(ServicesComponents.scala:60)
        at boot.AppPortalBuiltInComponentsFromContext.userService$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.userService(ProductionApplicationLoader.scala:21)
        at boot.modules.SecurityComponents.secure(SecurityComponents.scala:17)
        at boot.modules.SecurityComponents.secure$(SecurityComponents.scala:17)
        at boot.AppPortalBuiltInComponentsFromContext.secure$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.secure(ProductionApplicationLoader.scala:21)
        at boot.modules.WebComponents.cacheController(WebComponents.scala:38)
        at boot.modules.WebComponents.cacheController$(WebComponents.scala:38)
        at boot.AppPortalBuiltInComponentsFromContext.cacheController$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.cacheController(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.router$lzycompute(ProductionApplicationLoader.scala:29)
        at boot.AppPortalBuiltInComponentsFromContext.router(ProductionApplicationLoader.scala:29)
        at play.api.BuiltInComponents.httpRequestHandler(Application.scala:384)
        at play.api.BuiltInComponents.httpRequestHandler$(Application.scala:383)
        at play.api.BuiltInComponentsFromContext.httpRequestHandler$lzycompute(ApplicationLoader.scala:231)
        at play.api.BuiltInComponentsFromContext.httpRequestHandler(ApplicationLoader.scala:231)
        at play.api.BuiltInComponents.application(Application.scala:392)
        at play.api.BuiltInComponents.application$(Application.scala:386)
        at play.api.BuiltInComponentsFromContext.application$lzycompute(ApplicationLoader.scala:231)
        at play.api.BuiltInComponentsFromContext.application(ApplicationLoader.scala:231)
        at boot.modules.ProductionComponents.initializeDb(ProductionComponents.scala:51)
        at boot.modules.ProductionComponents.$init$(ProductionComponents.scala:43)
        at boot.AppPortalBuiltInComponentsFromContext.<init>(ProductionApplicationLoader.scala:22)
        at boot.ProductionApplicationLoader.load(ProductionApplicationLoader.scala:16)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:59)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:31)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
12:04:06.923 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: available, true
12:04:06.925 [main] DEBUG play.shaded.ahc.io.netty.util.internal.PlatformDependent0 - jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable
java.lang.IllegalAccessException: class play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6 cannot access class jdk.internal.misc.Unsafe (in module java.base) because module java.base does not export jdk.internal.misc to unnamed module @767a014e
        at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
        at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
        at java.base/java.lang.reflect.Method.invoke(Method.java:558)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0$6.run(PlatformDependent0.java:334)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:325)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:212)
        at play.shaded.ahc.io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:80)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:162)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetector.<init>(ResourceLeakDetector.java:207)
        at play.shaded.ahc.io.netty.util.ResourceLeakDetectorFactory$DefaultResourceLeakDetectorFactory.newResourceLeakDetector(ResourceLeakDetectorFactory.java:201)
        at play.shaded.ahc.io.netty.util.HashedWheelTimer.<clinit>(HashedWheelTimer.java:88)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.newNettyTimer(DefaultAsyncHttpClient.java:97)
        at play.shaded.ahc.org.asynchttpclient.DefaultAsyncHttpClient.<init>(DefaultAsyncHttpClient.java:87)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get$lzycompute(AhcWSModule.scala:60)
        at play.api.libs.ws.ahc.AsyncHttpClientProvider.get(AhcWSModule.scala:57)
        at play.api.libs.ws.ahc.AhcWSComponents.wsClient(AhcWSComponents.scala:32)
        at play.api.libs.ws.ahc.AhcWSComponents.wsClient$(AhcWSComponents.scala:29)
        at boot.AppPortalBuiltInComponentsFromContext.wsClient$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.wsClient(ProductionApplicationLoader.scala:21)
        at boot.modules.ExternalClientsComponents.authorisationServiceClient(ExternalClientsComponents.scala:49)
        at boot.modules.ExternalClientsComponents.authorisationServiceClient$(ExternalClientsComponents.scala:46)
        at boot.AppPortalBuiltInComponentsFromContext.authorisationServiceClient$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.authorisationServiceClient(ProductionApplicationLoader.scala:21)
        at boot.modules.ServicesComponents.userService(ServicesComponents.scala:60)
        at boot.modules.ServicesComponents.userService$(ServicesComponents.scala:60)
        at boot.AppPortalBuiltInComponentsFromContext.userService$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.userService(ProductionApplicationLoader.scala:21)
        at boot.modules.SecurityComponents.secure(SecurityComponents.scala:17)
        at boot.modules.SecurityComponents.secure$(SecurityComponents.scala:17)
        at boot.AppPortalBuiltInComponentsFromContext.secure$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.secure(ProductionApplicationLoader.scala:21)
        at boot.modules.WebComponents.cacheController(WebComponents.scala:38)
        at boot.modules.WebComponents.cacheController$(WebComponents.scala:38)
        at boot.AppPortalBuiltInComponentsFromContext.cacheController$lzycompute(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.cacheController(ProductionApplicationLoader.scala:21)
        at boot.AppPortalBuiltInComponentsFromContext.router$lzycompute(ProductionApplicationLoader.scala:29)
        at boot.AppPortalBuiltInComponentsFromContext.router(ProductionApplicationLoader.scala:29)
        at play.api.BuiltInComponents.httpRequestHandler(Application.scala:384)
        at play.api.BuiltInComponents.httpRequestHandler$(Application.scala:383)
        at play.api.BuiltInComponentsFromContext.httpRequestHandler$lzycompute(ApplicationLoader.scala:231)
        at play.api.BuiltInComponentsFromContext.httpRequestHandler(ApplicationLoader.scala:231)
        at play.api.BuiltInComponents.application(Application.scala:392)
        at play.api.BuiltInComponents.application$(Application.scala:386)
        at play.api.BuiltInComponentsFromContext.application$lzycompute(ApplicationLoader.scala:231)
        at play.api.BuiltInComponentsFromContext.application(ApplicationLoader.scala:231)
        at boot.modules.ProductionComponents.initializeDb(ProductionComponents.scala:51)
        at boot.modules.ProductionComponents.$init$(ProductionComponents.scala:43)
        at boot.AppPortalBuiltInComponentsFromContext.<init>(ProductionApplicationLoader.scala:22)
        at boot.ProductionApplicationLoader.load(ProductionApplicationLoader.scala:16)
        at play.core.server.ProdServerStart$.start(ProdServerStart.scala:59)
        at play.core.server.ProdServerStart$.main(ProdServerStart.scala:31)
        at play.core.server.ProdServerStart.main(ProdServerStart.scala)
```

I am not sure I mentioned it before, the application actually starts and run. I am just worried/wondering if this stacktrace is a sign that something is silently broken.

**Comment by mkurz** [comment]
@dadepo Check this out: https://github.com/netty/netty/issues/7817
This is not an error, but just a debug message which contains the stacktrace:

> It's not an error, it's a debug message with a stacktrace, change your logger level to something else apart from debug eg. info or warn.

That's also why your app runs smoothly, because it's not a bug.

Also see: https://stackoverflow.com/a/56226705/810109
Add something like this to your `logback.xml`:
```xml
<logger name="io.netty.util" level="ERROR"/>
```

**Comment by dwijnand** [comment]
Not all features of Java 9+ are implemented in Scala (even Scala 2.13), such as java modules.  Similarly about disabled reflection access.
